### Выбрать тест и проверить, есть  ли различия между выборками:
##### 1 )  Даны две  независимые выборки. Не соблюдается условие нормальности
x1  380,420, 290<br>
y1 140,360,200,900<br>
Сделайте вывод по результатам, полученным с помощью функции.

In [1]:
import numpy as np
from scipy import stats
x = np.array([380, 420, 290])
y = np.array([140, 360, 200, 900])

In [2]:
# используем критерий Манна–Уитни U:
# H0: μ1=μ2; H1: μ1≠μ2
stats.mannwhitneyu(x,y)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

pvalue=0.63: следовательно при а=0.05 (что гораздо < pvalue=0.63) **ПРИНИМАЕМ** нулевую гипотезу (H0: μ1=μ2) о том, что статистически значимых различий нет.

##### 2 ) Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?
1е измерение до приема препарата: 150, 160, 165, 145, 155<br>
2е измерение через 10 минут: 140, 155, 150,  130, 135<br>
3е измерение через 30 минут: 130, 130, 120, 130, 125

In [3]:
# используем критерий Фридмана "ХИ квадрат":
x1 = np.array([150, 160, 165, 145, 155])
x2 = np.array([140, 155, 150, 130, 135])
x3 = np.array([130, 130, 120, 130, 125])
stats.friedmanchisquare(x1, x2, x3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

pvalue=0.008: следовательно при а=0.05 (что > pvalue=0.008) **ОТВЕРГАЕМ** нулевую гипотезу о том, что статистически значимых различий нет и принимаем альтернативную гипотезу, о том, что статистически значимые различия между группами обнаружены.

##### 3 ) Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

In [4]:
# используем критерий Уилкоксона
x1 = np.array([150, 160, 165, 145, 155])
x2 = np.array([140, 155, 150, 130, 135])
stats.wilcoxon(x1, x2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

pvalue=0.06: следовательно при а=0.05 (что < pvalue=0.06)  **ПРИНИМАЕМ** нулевую  гипотезу о том, что статистически значимых различий нет.

##### 4) Даны 3 группы  учеников плавания.
В 1 группе время на дистанцию 50 м составляют: 56, 60, 62, 55, 71, 67, 59, 58, 64, 67<br>
Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53<br>
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

In [5]:
# используем критерий Крускала-Уоллиса H:
x1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
x2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
x3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])
stats.kruskal(x1, x2, x3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

pvalue=0.065: следовательно при а=0.05 (что < pvalue=0.065)  **ПРИНИМАЕМ** нулевую  гипотезу о том, что статистически значимых различий нет.

##### 5) Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [6]:
X = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
m=2.5
n=10
a=0.05
x=X.mean()
x

2.5279999999999996

In [7]:
s=X.std(ddof=1)
s

0.1572542173961923

In [8]:
t0 = (x-m)/(s/np.sqrt(n))
t0

0.5630613661802959

In [9]:
t1 = stats.t.ppf(1-a, n-1) #с доверительной вероятностью 0,95 и степенью свободы 9 (односторонний тест)
t1

1.8331129326536335

In [10]:
t3 = stats.t.ppf(a, n-1) #с доверительной вероятностью 0,05 и степенью свободы 9 (односторонний тест)
t3

-1.8331129326536337

In [11]:
t2 = stats.t.ppf(1-a/2, n-1) #с доверительной вероятностью 0,95 (для 0.975) и степенью свободы 9 (двусторонний тест)
t2

2.2621571627409915

1. H0 -  партия изготавливается со средним арифметическим 2,5 см, H1 - партия изготавливается со средним арифметическим НЕ 2,5 см;
2. дисперсия генеральной выборки неизвестна - используем t-критерий;
3. t-расчетное = 0.563, т.е x=2.53 лежит от m=2.5 в 0.563 сигмах (стандартных ошибках среднего(SE)) вправо;
4. t-табличное для 1-а=0.95 (симетрично a=0.05) для 9 степеней свободы лежит в 1.833 сигмах вправо;
5. т.е. t-расчетное 0.563 < t-табличное 1.833, и лежит в области нулевой гипотизы. При а=0.05 **ПРИНИМАЕМ** нулевую гипотезу H0.